lib/Interpreter/CIFactory.cpp

```diff
@@ -297,8 +318,18 @@ namespace {
 
     #if defined(__GLIBCXX__)
       // Avoid '__float128 is not supported on this target' errors
-      if (!opts.Language && !opts.StdVersion)
-        sArguments.addArgument("-std=c++11");
+      if (!opts.Language && !opts.StdVersion) {
+        switch (CxxStdCompiledWith()) {
+          case 17:
+            // Hopefully -D__float128=void can be removed when -std=c++17 works.
+            sArguments.addArgument("-D__float128=void");
```

Comment:
Thanks. Then lets see what Travis says without that line, and if it barfs we see how to fix it. We should at least find or file a clang bug in that case.